Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -45,27 +45,40 @@ def fix_dynamic_linkage
         change_dylib_id(dylib_id_for(file), file) if file.dylib?
 
         each_linkage_for(file, :dynamically_linked_libraries) do |bad_name|
-          # Don't fix absolute paths unless they are rooted in the build directory
-          next if bad_name.start_with?("/") &&
-                  !rooted_in_build_directory?(bad_name)
-
-          new_name = fixed_name(file, bad_name)
-          change_install_name(bad_name, new_name, file) if new_name != bad_name
+          # Don't fix absolute paths unless they are rooted in the build directory.
+          new_name = if bad_name.start_with?("/") && !rooted_in_build_directory?(bad_name)
+            bad_name
+          else
+            fixed_name(file, bad_name)
+          end
+          loader_name = loader_name_for(file, new_name)
+          change_install_name(bad_name, loader_name, file) if loader_name != bad_name
         end
 
         each_linkage_for(file, :rpaths) do |bad_name|
           # Strip duplicate rpaths and rpaths rooted in the build directory.
-          next if !rooted_in_build_directory?(bad_name) &&
-                  (file.rpaths.count(bad_name) == 1)
-
-          delete_rpath(bad_name, file)
+          if rooted_in_build_directory?(bad_name) || (file.rpaths.count(bad_name) > 1)
+            delete_rpath(bad_name, file)
```

Comment:
`file.rpaths` has resolved `@loader_path/` leading to `julia` issue. Printing out the data:
```
["/usr/local/Cellar/julia/1.9.2/lib/julia",
 "/usr/local/opt/curl/lib",
 "/usr/local/opt/mbedtls@2/lib",
 "/usr/local/opt/openblas/lib",
 "/usr/local/opt/gcc/lib/gcc/current",
 "/usr/local/lib",
 "/usr/lib",
 "/usr/local/Cellar/julia/1.9.2/lib/julia"]
==> Deleting rpath /usr/local/Cellar/julia/1.9.2/lib/julia in /usr/local/Cellar/julia/1.9.2/lib/julia/libcholmod.3.0.14.dylib
```

Reply:
> file.rpaths has resolved @loader_path/ leading to julia issue where earlier RPATH is deleted changing which dylib gets loaded

@cho-m Yeah, that's bad and is caused by the changes in this PR (specifically, https://github.com/Homebrew/brew/pull/15571/commits/15a0c7fd7df2cae55e3b702a04e7bbf1ad401f27). 

Changing the ``files.rpaths.count(bad_name)`` call to ``files.rpaths(resolve_variable_references: false).count(bad_name)`` should go back to the old behavior, which I think is the correct behavior here (@carlocab please correct me if I'm wrong).

Reply:
Yes, Julia should be fixed now.

Reply:
Right, fix is in #15722. Thanks!